Tickets/dm 15300

menanteau opened the pull request.

menanteau committed:
Updates to the bash script that start the web server for the HeaderService.
It now gets the correct PID and it compatible with boty Python2/3.

ktlim commented:
You may want to consider https://developer.lsst.io/work/flow.html#appendix-commit-organization-best-practices
`git add -p` can be very useful to separate out conceptually different modifications into distinct commits.  Any time your commit message says "in addition" or "and", it might be an indication that this should be two commits.  (But we all combine at times.)

You might also consider https://developer.lsst.io/work/flow.html#appendix-commit-message-best-practices
In particular, use of the imperative and 50 character initial lines seem to be missing here.

ktlim commented:
You should never be merging master into a tickets branch.  You should be rebasing instead.

menanteau commented:
That was not the intention, I though that I was doing the opposite!

ktlim commented on bin/DMHS_ATS_configurable:
```diff
@@ -31,29 +30,30 @@ def cmdline():
     parser.set_defaults(**conf_defaults)
     args = parser.parse_args(args=remaining_argv)
 
-    #print ("Will use:")
```
Debugging code can often be left in, rather than commented out, with output only at DEBUG level.

Also, we typically do not use a space between `print` and the parenthesis, as `print` is a function call.  This needs to be fixed throughout.

menanteau replied:
Cleaned this up.

ktlim commented on bin/LFO_dmHeaderService_EventLogger:
```diff
@@ -26,7 +25,7 @@ if __name__ == "__main__":
             LOGGER.info("Version: %s" % event.version)
             LOGGER.info("priority: %s" % event.priority)
-            retval==-1
+            retval == -1
```
`==` is probably not what is desired.

menanteau replied:
Changed to `retval = -1`

ktlim commented on bin/assemble_FITS:
```diff
@@ -35,60 +35,63 @@ def cmdline():
     if naxis1 == 0 or naxis2 == 0:
-        print("WARNING: Both NAXIS1/NAXIS2 muste be non-zero, current values are:{},{}".format(naxis1,naxis2))
-    
+        print("WARNING: Both NAXIS1/NAXIS2 muste be non-zero, \
```
Backslash continuation within a quoted string is unusual and probably does not have the effect desired (particularly with regard to spacing).  Concatenation is more typical.

menanteau replied:
Fixed

ktlim commented on python/HeaderService/camera_coords.py:
```diff
@@ -1,6 +1,10 @@
 import HeaderService
+import math
+
+"""This modules describe the CCD/Camera coordinate systems as described in
```
Here and elsewhere, Python files need to have the LSST license header, and docstrings should conform to the standards in https://developer.lsst.io/python/numpydoc.html

menanteau replied:
If understand this correctly, the string is correctly formatted, but I'm missing the License? If that is the case, I'd like to create a new separate ticket to add the License to all relevant files in both salpytools and headerService.

timj replied:
"This module" not "This modules"

menanteau replied:
Typo had been fixed too. Thanks for being an attentive reader.

menanteau replied:
I've created a ticket to add the LSST license header as a separate task, DM-15861.

menanteau committed:
Updates following end of code review

menanteau committed:
Intermediate commit to bring code to PEP8 standards.

In addition obsolete code (i.e. previous version of the HeaderService) that are no longer supported due to changes in the XML defintion and the behaviour of the CSC have been removed.

Further clean up of obsolete script and fixes to make code PEP8 compliant following flake8 suggestions.
